### PR TITLE
Simplify Role handling, eliminating RolesController.

### PR DESCRIPTION
**Pull request by jnaviask**
## Link to Issue
Closes: #8285

## Description of Changes
The `app.roles` controller maintained a `roles` list, which functioned as a unclear abstraction: it contained "role" objects for the current user only. Given that a role object is a simple permission string on top of an account object (as membership is determined by address+community_id on the account), but is functionally an entirely separate object, the goal of this PR was to eliminate all use of Roles on the client. 

Removing Roles involved enforcing two main contracts:
1. Communities must own their own list of mods and admins, as a list of addresses.
2. The UserStore must own the list of accounts owned by the currently logged in user, i.e. the canonical list of address x community pairings.

Given these two contracts, it should be straightforward to check a role at any time, reflected in the updates to the Permissions util.

The remaining work revolved around updating admin and moderator management / role assignment within a community. There are still outstanding product questions in this area, as we display profile names but ultimately upgrade individual addresses.

Finally, a change to remove "includeRoles" was made to searchProfiles, as it was no longer needed. Legacy Role objects and types were stripped from various places across the codebase alongside this change.

## Test Plan
- Create community
- Update community (http://localhost:8080/:communityId/manage/profile)
- Check that the admins and members management page works, you should be able to remove and add members and they should persist on page refresh (http://localhost:8080/:communityId/manage/moderators)
- Check that the you can create/update/delete groups (http://localhost:8080/:communityIdmembers/groups/create)

### CODE REVIEW

**Review by kurtisassad — DISMISSED** [review]
Just approving this to get it out of my PR review queue, but this should require 2 other approvals since I have a conflict of interest since I had worked on it.

File: libs/model/src/models/address.ts
Comment (masvelio) [inline libs/model/src/models/address.ts:104]: let's resolve this TODO by adding the index or removing the comment

File: packages/commonwealth/client/scripts/controllers/app/login.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/controllers/app/login.ts:95]: i remember that this piece of code was necessary in the situation when we wanted to persist the default address selected when you reload the page/open the app again. If this still works, then fine

File: packages/commonwealth/client/scripts/utils/Permissions.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/utils/Permissions.ts:21]: makes sense

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_sidebar_menu.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/cw_sidebar_menu.tsx:133]: I think this should not be deleted?

**Review by masvelio — COMMENTED** [review]
it would be great to resolve all those in-code comments and TODO's before merging. Great you caught those inconsistencies. Will do some CA testing

File: libs/model/src/models/address.ts
Comment (jnaviask) [reply libs/model/src/models/address.ts:104]: It really should be an XXX or note of some sort, we don't want to modify the schema unless there's performance issues querying mods/admins, but we also don't want to lose the idea that we can simply add an index... I will update it.

File: packages/commonwealth/client/scripts/controllers/app/login.ts
Comment (jnaviask) [reply packages/commonwealth/client/scripts/controllers/app/login.ts:95]: The /setDefaultRole call should be enough -- updating the roles controller should affect the *current* active address, and not the active address on reload. That said, I will test this functionality now.

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_sidebar_menu.tsx
Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_sidebar_menu.tsx:133]: It's irrelevant here -- this is a shouldShowAvatarOnly User (i.e. the little icons in the sidebar menu) and shouldShowAsDeleted only affects the text output ("Anonymous" vs "Deleted").

**Comment by jnaviask** [comment]
@masvelio I have addressed your comments -- lmk if these answers suffice.

File: packages/commonwealth/client/scripts/controllers/app/login.ts
Comment (jnaviask) [reply packages/commonwealth/client/scripts/controllers/app/login.ts:95]: On testing, the active address persists both locally and across refreshes, so we should be OK here.

Comment (masvelio) [reply packages/commonwealth/client/scripts/controllers/app/login.ts:92]: thanks for checking

**Review by masvelio — APPROVED** [review]
1. We have significant change how the app is functioning right now. Previously, when you were admin of community with address A and you joined with address B, address B was just a user. Video below

https://github.com/user-attachments/assets/0895140b-c653-4f49-8a58-7a86f5c72f7e

After your changes, both addresses are admins automatically. 

https://github.com/user-attachments/assets/52919671-7396-4436-81d3-a8fc488c9f06

Is this expected behaviour? 

2. When I make a regular user an admin, user gets removed from the members list. In my case sometimes immediately and sometimes after the refresh. Members list should be persistent, only admins and mods should be changeable. If this is not a regression, would be good to create a ticket for the bug. Or, we can create a ticket for looking into this page as a whole, as the logic there for handling the lists etc is very messy (cleanup ticket). 

https://github.com/user-attachments/assets/2c04baa4-9b33-4e2f-baaa-1aaf60c6984c


Apart from those two things I didn't encounter anymore problems so after resolving my two questions ☝️ it's g2g from my side.

File: packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx:54]: yes

File: packages/commonwealth/server/controllers/server_communities_methods/update_community.ts
Comment (mzparacha) [inline packages/commonwealth/server/controllers/server_communities_methods/update_community.ts:213]: did we verify this? We allow both community/site admin to update community data in frontend

File: packages/commonwealth/server/controllers/server_communities_methods/update_community.ts
Comment (mzparacha) [inline packages/commonwealth/server/controllers/server_communities_methods/update_community.ts:1]: while checking this i couldn't get `pnpm set-super-admin` to work

File: packages/commonwealth/client/scripts/views/pages/create_community/community_input_rows.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/create_community/community_input_rows.tsx:21]: `skipDeinitChain` can be removed from `app`

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:56]: if `0` index contains a different address connected to a different community, then it might break. We should choose address of the active community (if there are multiple then maybe we display them individually)

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:89]: same as [this](https://github.com/hicommonwealth/commonwealth/pull/8509/files#r1691535063)

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:98]: same https://github.com/hicommonwealth/commonwealth/pull/8509/files#r1691535063

File: packages/commonwealth/client/scripts/models/ChainInfo.ts
Comment (jnaviask) [reply packages/commonwealth/client/scripts/models/ChainInfo.ts]: TODO is incorrect, removing

File: packages/commonwealth/client/scripts/views/pages/create_community/community_input_rows.tsx
Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/pages/create_community/community_input_rows.tsx:21]: Done

**Review by timolegros — APPROVED** [review]
Went through test plan without issues and server changes LGTM.

**Comment by jnaviask** [comment]
@masvelio 
> Previously, when you were admin of community with address A and you joined with address B, address B was just a user... After your changes, both addresses are admins automatically. ... Is this expected behavior?

We have no guidelines from product around what behavior is expected or unexpected, because we have no actual definition about what should happen wrt a solid model of how accounts should function. My personal gut-check is it's better for a User to be admin if any of their addresses is an admin in a community, rather than them being confused why they can't access admin functionality without switching accounts, but again, we have no clear decision either way here from the left.

> When I make a regular user an admin, user gets removed from the members list. In my case sometimes immediately and sometimes after the refresh. Members list should be persistent, only admins and mods should be changeable. If this is not a regression, would be good to create a ticket for the bug. Or, we can create a ticket for looking into this page as a whole, as the logic there for handling the lists etc is very messy (cleanup ticket).

This is specifically implemented behavior, as the members list previously only displayed non-admins (see: UpgradeRolesForm.tsx), but we can change it if we think that it will be a better product experience to show all members.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:46]: We still check for moderators, it is basically all members that are not admin (members and moderators)

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:56]: I think the problem here is that we need to display a non-admin address for the user to associate that user with some address to display them on the members page:
image

The issue is that in this case, a user can be associated with multiple addresses, so it is not obvious how to handle this members list. The current approach of this PR is just to leave it as is before.

But if we want to resolve this in a better way, how should it be handled? Maybe the profile can be linked multiple times, once for each of their addresses? Also maybe it should be handled in a followup PR since it is a bit out of the scope of the roles removal.

Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx:56]: Ok seems like it is part of a separate ticket I am assigned to. I will implement this functionality in a followup PR:
https://github.com/hicommonwealth/commonwealth/issues/8524

File: packages/commonwealth/server/controllers/server_communities_methods/update_community.ts
Comment (kurtisassad) [reply packages/commonwealth/server/controllers/server_communities_methods/update_community.ts:213]: yes this is correct, remove the question part of the comment

File: packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx]: done

**Review by mzparacha — APPROVED** [review]

**Pull request merged**
